src/app/SIPUserAgents/ISIPServerUserAgent.cs

```diff
@@ -18,6 +18,8 @@ namespace SIPSorcery.SIP.App
 {
     public delegate void SIPUASDelegate(ISIPServerUserAgent uas);
 
+    public delegate void SIPUASCancelDelegate(ISIPServerUserAgent uas, SIPRequest sipCancelRequest);

```

Comment:
I will update this to not expose the SIPRequest, I will instead expose "string reason" and "string[] customHeaders"